Our (gcc/binutils) toolchain doesn't do ARM targets, hence we will get
"unable to interface with target machine" errors while building if we
attempt to.  Disable those targets.

Patch by Reza Jelveh from
https://github.com/fishman/timebomb-gentoo-osx-overlay/blob/master/sys-devel/llvm/files/llvm-3.4-fix_darwin_build.patch

--- a/projects/compiler-rt/make/platform/clang_darwin_embedded.mk
+++ b/projects/compiler-rt/make/platform/clang_darwin_embedded.mk
@@ -27,20 +27,20 @@ UniversalArchs :=
 # Soft-float version of the runtime. No floating-point instructions will be used
 # and the ABI (out of necessity) passes floating values in normal registers:
 # non-VFP variant of the AAPCS.
-Configs += soft_static
-UniversalArchs.soft_static := armv6m armv7m armv7em armv7
+# Configs += soft_static
+# UniversalArchs.soft_static := armv6m armv7m armv7em armv7
 
 # Hard-float version of the runtime. On ARM VFP instructions and registers are
 # allowed, and floating point values get passed in them. VFP variant of the
 # AAPCS.
 Configs += hard_static
-UniversalArchs.hard_static := armv7em armv7 i386 x86_64
+UniversalArchs.hard_static := i386 x86_64
 
-Configs += soft_pic
-UniversalArchs.soft_pic := armv6m armv7m armv7em armv7
+# Configs += soft_pic
+# UniversalArchs.soft_pic := armv6m armv7m armv7em armv7
 
 Configs += hard_pic
-UniversalArchs.hard_pic := armv7em armv7 i386 x86_64
+UniversalArchs.hard_pic := i386 x86_64
 
 CFLAGS := -Wall -Werror -Oz -fomit-frame-pointer -ffreestanding
 
--- a/tools/clang/runtime/compiler-rt/Makefile
+++ b/tools/clang/runtime/compiler-rt/Makefile
@@ -85,7 +85,7 @@ RuntimeLibrary.darwin.Configs := \
 	profile_osx.a profile_ios.a \
 	ubsan_osx.a
 RuntimeLibrary.darwin_embedded.Configs := \
-	soft_static.a hard_static.a soft_pic.a hard_pic.a
+	 hard_static.a  hard_pic.a
 
 # Support building compiler-rt with relocatable SDKs.
 #
